test/functional/services/table.ts

```diff
@@ -17,25 +17,38 @@
  * under the License.
  */
 
-export function TableProvider({ getService }) {
+import { FtrProviderContext } from '../ftr_provider_context';
+import { WebElementWrapper } from './lib/web_element_wrapper';
+
+export function TableProvider({ getService }: FtrProviderContext) {
   const testSubjects = getService('testSubjects');
-  // const retry = getService('retry');
 
   class Table {
+    /**
+     * Finds table and returns data in the nested array format
+     * @param dataTestSubj data-test-subj selector
+     */
 
-    async getDataFromTestSubj(testSubj) {
-      const table = await testSubjects.find(testSubj);
+    public async getDataFromTestSubj(dataTestSubj: string): Promise<string[][]> {
+      const table = await testSubjects.find(dataTestSubj);
       return await this.getDataFromElement(table);
     }
 
-    async getDataFromElement(table) {
-      // Convert the data into a nested array format:
-      // [ [cell1_in_row1, cell2_in_row1], [cell1_in_row2, cell2_in_row2] ]
-      const rows = await table.findAllByTagName('tr');
-      return await Promise.all(rows.map(async row => {
-        const cells = await row.findAllByTagName('td');
-        return await Promise.all(cells.map(async cell => await cell.getVisibleText()));
-      }));
+    /**
+     * Converts the table data into nested array
+     * [ [cell1_in_row1, cell2_in_row1], [cell1_in_row2, cell2_in_row2] ]
+     * @param table
+     */
+    public async getDataFromElement(table: WebElementWrapper): Promise<string[][]> {
+      const $ = await table.parseDomContent();
+      return $('tr')
+        .toArray()
+        .map(tr =>
+          $(tr)
+            .find('td')
+            .toArray()
+            .map(cell => $(cell).text())
+        );
```

Comment:
Using cheerio here as well since we already doing it for table parsing in another service.